### ISSUE TITLE
False positive in `undesirable_function_linter()`

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2584]
`message` is a parameter name, but it's mistaken to be the function name:

``` r
library(lintr)

f <- function(message) print(message)
f("hi")
#> [1] "hi"

lint(
  text = "f <- function(message) print(message)", 
  linters = undesirable_function_linter(modify_defaults(
    defaults = default_undesirable_functions,
    message = "use cli::cli_inform()"
  ))
)
#> <text>:1:30: style: [undesirable_function_linter] Avoid undesirable function "message". As an alternative, use cli::cli_inform().
#> f <- function(message) print(message)
#>                              ^~~~~~~
```

<sup>Created on 2024-05-25 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by IndrajeetPatil** [comment]
IINM, we should be able to distinguish between these two cases by checking if it's `SYMBOL_FUNCTION_CALL` versus `SYMBOL_FORMALS`.

**Comment by MichaelChirico** [comment]
but that's not SYMBOL_FORMALS, it's SYMBOL. you can ignore it with symbol_is_undesirable=FALSE.

**Comment by MichaelChirico** [comment]
note that in the same function you could do sapply(strs, message) and it would be the "undesirable" function applied, not the function argument. so going down the route of "try and look in parent for matching SYMBOL_FORMALS" is very error prone for many reasons.

**Comment by IndrajeetPatil** [comment]
You are right, forgot about this argument!
``` r
library(lintr)

lint(
  text = "f <- function(message) print(message)", 
  linters = undesirable_function_linter(modify_defaults(
    defaults = default_undesirable_functions,
    message = "use cli::cli_inform()"
  ), symbol_is_undesirable = FALSE)
)
```

<sup>Created on 2024-05-25 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>
